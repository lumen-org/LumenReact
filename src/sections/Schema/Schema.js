import React, {Component} from "react";
import TitelH1 from "../../components/Titles/TitelH1";
import Field from "../../components/Fields/Field";
import TitelH2 from "../../components/Titles/TitleH2";

class Schema extends Component {
  constructor(props) {
    super(props);
    this.state = {
      datalist1: [1,3, 2,3,4,5,6,9],
      dataList2: [3,4,5,6,87,1,10]
    }
  }
  render() {
    return (
        <div className="Schema">
          <TitelH1 value={"Schema"}/>
          <Field title={"Quantitative"} data={this.state.datalist1}/>
          <Field title={"Categorical"} data={this.state.dataList2}/>
          <TitelH2 value={"Drop here to remove"} />
        </div>
    );
  }
}

export default Schema;
