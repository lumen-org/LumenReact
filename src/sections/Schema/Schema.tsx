import React, { Component } from "react";
import PropTypes from "prop-types";
import TitleH1 from "../../components/Titles/TitleH1";
import Field from "../../components/Field";
import "./Schema.css";

type SchemaProps = {
  quantitative: object,
  categorical: object
}

/**
 * lets talk about this next week:
 * https://stackoverflow.com/questions/41746028/proptypes-in-a-typescript-react-application
 */
class Schema extends Component<SchemaProps> {
  static propTypes = {
    quantitative: PropTypes.array.isRequired,
    categorical: PropTypes.array.isRequired
  };

  render() {
    const { quantitative, categorical } = this.props;
    return (
      <div className="Schema">
        <TitleH1 value={"Schema"} />
        <Field title={"Quantitative"} data={quantitative} dropable={false} />
        <Field title={"Categorical"} data={categorical} dropable={false} />
      </div>
    );
  }
}

export default Schema;
