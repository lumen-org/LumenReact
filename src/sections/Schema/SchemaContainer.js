import React from "react";
import { connect } from "react-redux";
import Schema from "./Schema";
import { selectSchemeNames } from "../../states/schemes/selector";

class SchemaContainer extends React.Component {
  render() {
    const { schemeNames } = this.props;
    return (
      <div>
        {this.props.activeSchema !== -1 && (
          <Schema
            quantitative={schemeNames.quantitative}
            categorical={schemeNames.categorical}
          />
        )}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    schemeNames: selectSchemeNames(state),
  };
};

export default connect(mapStateToProps, null)(SchemaContainer);
