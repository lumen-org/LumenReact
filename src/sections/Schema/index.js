import React, { Component } from "react";

class Schema extends Component {
  render() {
    return <p> Schema</p>;
  }
}

export default Schema;
