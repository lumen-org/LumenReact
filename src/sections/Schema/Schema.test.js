import React from "react";
import Schema from "./Schema";
import TestRenderer from "react-test-renderer";

// https://jestjs.io/docs/en/tutorial-react

jest.mock("../../components/Field", () => (props) => (<div>
    <div>
        {props.title}
    </div>
    <div>
        {props.data}
    </div>
</div>));

let spy = null;
beforeEach(() => {
  spy = jest.spyOn(console, 'error').mockImplementation();
});

afterEach(() => {
  // cleanup on exiting
  spy.mockRestore();
});

describe("Schema functionality test", () => {
    test("Schema displays dimensions correctly", () => {
        const quantitative = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "species"];
        const categorical = ["sepal_length", "sepal_width", "petal_length", "petal_width"];
    
        const component = TestRenderer.create(
            <Schema quantitative={quantitative} categorical={categorical} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    test("Schema throws no exception when all attributes are passed correctly", () => {
        const quantitative = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "species"];
        const categorical = ["sepal_length", "sepal_width", "petal_length", "petal_width"];
    
        TestRenderer.create(
            <Schema quantitative={quantitative} categorical={categorical} />
        );
        expect(console.error).not.toBeCalled();
    });
});

describe("Schema Test Errors thrown", () => {
    test("Schema throws exception when no quantitative attribute is passed", () => {
        const categorical = ["sepal_length", "sepal_width", "petal_length", "petal_width"];
        TestRenderer.create(
            <Schema categorical={categorical} />
        );
        expect(console.error).toBeCalled();
    });
    
    test("Schema throws exception when no categorical attribute is passed", () => {
        const quantitative = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "species"];
        TestRenderer.create(
            <Schema quantitative={quantitative} />
        );
    
        expect(console.error).toBeCalled();
    });
    // not working 
    /*
    test("Schema throws exception when no attribute is passed", () => {
        jest.spyOn(console, "error");
        TestRenderer.create(
            <Schema />
        );
        expect(console.error).toBeCalled();
    })
    */ 
})



// My proposel evtl.
// beforeEach(() => {
// jest.spyOn(console, 'error')
// jest.spyOn(console, 'warn')
// })
// afterEach(() => {
// /* eslint-disable no-console,jest/no-standalone-expect */
// expect(console.error).not.toBeCalled()
// expect(console.warn).not.toBeCalled()
// })
