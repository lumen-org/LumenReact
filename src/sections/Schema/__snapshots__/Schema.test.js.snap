// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema displays dimensions correctly 1`] = `
<div
  className="Schema"
>
  <div
    className="title-h1"
  >
    Schema
  </div>
  <div>
    <div>
      Quantitative
    </div>
    <div>
      1
      2
      3
      4
      5
      6
      7
      8
      9
      10
      11
      12
      13
      14
      15
      16
      species
    </div>
  </div>
  <div>
    <div>
      Categorical
    </div>
    <div>
      sepal_length
      sepal_width
      petal_length
      petal_width
    </div>
  </div>
</div>
`;
